cachi2/core/package_managers/npm.py

```diff
@@ -140,16 +162,43 @@ class PackageLock:
 
     def __init__(self, lockfile_path: RootedPath, lockfile_data: dict[str, Any]) -> None:
         """Initialize a PackageLock."""
+        self._workspaces: list[str] = []
         self._lockfile_path = lockfile_path
         self._lockfile_data = lockfile_data
         self._packages = self._get_packages()
         self._dependencies = self._get_dependencies()
 
+    @property
+    def packages(self) -> list[Package]:
+        """Get list of packages loaded from package-lock.json."""
+        return self._packages
+
+    @property
+    def lockfile_data(self) -> dict[str, Any]:
+        """Get content of package-lock.json stored in Dictionary."""
+        return self._lockfile_data
+
     @property
     def lockfile_version(self) -> int:
         """Get the lockfileVersion from package-lock.json data."""
         return self._lockfile_data["lockfileVersion"]
 
+    def _check_if_package_is_workspace(self, resolved_url: str) -> bool:
+        """Test if package is workspace based on main package workspaces."""
+        if (
+            "packages" not in self._lockfile_data
+            or "" not in self._lockfile_data["packages"]
+            or "workspaces" not in self._lockfile_data["packages"][""]
+        ):
+            return False
+
+        main_package_workspaces = self._lockfile_data["packages"][""]["workspaces"]
+
+        for main_package_workspace in main_package_workspaces:
+            if fnmatch.fnmatch(resolved_url, main_package_workspace.removeprefix("./")):
+                return True
```

Comment:
Rather than `removeprefix("./")`, could we use `Path(...).as_posix()`? There's a lot of non-normalized ways to write a file path, redundant `./` is just one of many things that pathlib could handle for us

Reply:
Fixed